src/lib/api.ts

```diff
@@ -0,0 +1,98 @@
+import * as TE from "fp-ts/TaskEither";
+import { identity } from "fp-ts/lib/function";
+
+const API_URL =
+  process.env.NODE_ENV === "production"
+    ? "s://idk-the-adress-yet"
+    : "://localhost:8080";
+
+export type Message = {
+  content: string;
+  id: string;
+};
+export type Chat = {
+  title: string;
+  id: string;
+  adminToken: string;
+  messages: Message[];
+};
+
+export const createChatTE = (title: string, chatID: string) =>
+  TE.tryCatchK(
+    () =>
+      fetch(`http${API_URL}/chats`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ title, id: chatID }),
+      }).then((res) =>
+        res.ok ? (res.json() as Promise<Chat>) : Promise.reject("")
```

Comment:
Tady bych možná použil https://gcanti.github.io/fp-ts/modules/Json.ts.html na to parsování. Nebo ještě líp udělal bych si openapi specifikaci a vygeneroval si klienta